include/experimental/__p0009_bits/utility.hpp

```diff
@@ -64,6 +66,112 @@ constexpr struct
   }
 } stride;
 
+template<class T>
+MDSPAN_INLINE_FUNCTION
+constexpr void maybe_unused_variable(const T&) {}
+
+
+// same as std::integral_constant but with __host__ __device__ annotations on
+// the implicit conversion function and the call operator
+template <class T, T v>
+struct integral_constant {
+  using value_type         = T;
+  using type               = integral_constant<T, v>;
+
+  MDSPAN_INLINE_FUNCTION_DEFAULTED
+  constexpr integral_constant() = default;
+  
+  MDSPAN_INLINE_FUNCTION_DEFAULTED
+  constexpr integral_constant(std::integral_constant<T,v>) {};
```

Comment:
`DEFAULTED` seems inappropriate. Do you need the extra annotations at all here?
```suggestion
  MDSPAN_INLINE_FUNCTION
  constexpr integral_constant(std::integral_constant<T,v>) {};
```

Reply:
Can't be host device in the 1st place

Reply:
yes it can. Nothing in this calls device functions. Constructor of std::integral_constant is defaulted, so it works too. The only thing which doesn't work of std::integral_constant is the conversion operator to its scaler type.

Reply:
Then what Daniel said

Reply:
(minus the inline)

Reply:
yeah clear bug (funnily enough one of the primary bugs we had elsewhere)